source/Octopus.Tentacle/Maintenance/WorkspaceCleanerTask.cs

```diff
@@ -2,84 +2,29 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Octopus.Diagnostics;
+using Octopus.Tentacle.Background;
 
 namespace Octopus.Tentacle.Maintenance
 {
-    public interface IWorkspaceCleanerTask
-    {
-        void Start();
-        void Stop();
-    }
+    public interface IWorkspaceCleanerTask : IBackgroundTask
+    {}
 
-    class WorkspaceCleanerTask : IWorkspaceCleanerTask, IDisposable
+    class WorkspaceCleanerTask : BackgroundTask, IWorkspaceCleanerTask
     {
         readonly WorkspaceCleanerConfiguration configuration;
         readonly WorkspaceCleaner workspaceCleaner;
-        readonly ISystemLog log;
-
-        readonly CancellationTokenSource cancellationTokenSource = new ();
-        readonly object taskLock = new();
-
-        Task? cleanerTask;
 
         public WorkspaceCleanerTask(
             WorkspaceCleanerConfiguration configuration,
             WorkspaceCleaner workspaceCleaner, 
-            ISystemLog log)
+            ISystemLog log) : base(log, TimeSpan.FromSeconds(30))
         {
             this.configuration = configuration;
             this.workspaceCleaner = workspaceCleaner;
-            this.log = log;
-        }
-
-        public void Dispose()
-        {
-            Stop();
-
-            cancellationTokenSource.Dispose();
-        }
-
-        public void Start()
-        {
-            lock (taskLock)
-            {
-                if (cleanerTask is not null)
-                {
-                    log.Error("Workspace cleaner task already running.");
-                    return;
-                }
-
-                cleanerTask = Task.Run(RunTask);
-            }
         }
 
-        public void Stop()
+        protected override async Task RunTask(CancellationToken cancellationToken)
         {
-            lock (taskLock)
-            {
-                if (cleanerTask is null) return;
-
-                try
-                {
-                    cancellationTokenSource.Cancel();
-
-                    cleanerTask.Wait();
```

Comment:
I noticed this currently has an infinite wait. After the refactor, this becomes a 30s wait which I think is more reasonable.